R/variables.R

```diff
@@ -250,4 +250,38 @@ Probabilities.GradientBoost <- function(object)
     return(as.matrix(probabilities))
 }
 
-
+#' @title Compute the Propensity Weight scores for a binary classification model
+#' @description Computes the propensity weights from a binary classification model
+#'   using the inverse predicted probabilities of each observation being in the
+#'   positive class.
+#' @param object A \code{Regression} or \code{MachineLearning} output.
+#' @return A vector of numeric propensity weights for each case in the model.
+#' @importFrom flipU OutcomeVariable
+#' @export
+PropensityWeights <- function(object)
+{
+    is.binary.ML.model <- inherits(object, "MachineLearning") && !object$numeric.outcome
+    is.binary.Regression <- inherits(object, paste0(c("Binary", "Multinomial"), "LogitRegression"))
+    binary.outcome.msg <- paste0("Propensity Weights can only be saved for binary classification models ",
+                                 "(e.g. Binary Logit Regression or a Machine Learning models with a ",
+                                 "binary outcome variable)")
+    if(!(is.binary.ML.model || is.binary.Regression))
+        stop(binary.outcome.msg)
+    probabilities <- Probabilities(object)
+    n.classes <- NCOL(probabilities)
+    if (n.classes > 2L)
+        stop("The supplied model is a multiclass classification model with ", n.classes, " ",
+             "outcome categories/class labels. Computing Propensity weights is only supported ",
+             "for binary classification models. E.g. Binary Logit, or a Machine Learning model ",
+             "with a Binary variable as the outcome variable. Consider merging categories ",
+             "in the outcome variable to produce a binary classification before computing ",
+             "propensity weights.")
+    outcome.variable <- OutcomeVariable(object[["formula"]], object[["model"]])
+    if (n.classes != 2L)
```

Comment:
Why is this needed if you've already checked if `n.classes > 2L`? How could `n.classes` be anything other than 2 given the checks on L268 and L272?

Reply:
Its a weird edge case but still possible. It can occur that a binary logit model can be created where all the outcome observations are of the same class. The user is simply supplied with a warning in that case. That probably shouldn't just warn and it should error instead. For example, 
![image](https://user-images.githubusercontent.com/4605715/129511217-83c8aad4-5654-4816-9e2e-315a2c22d397.png)

Reply:
By same class I mean the outcome variable is a factor with only one level.